custom_components/tuya_local/devices/madimack_elite_v3_heatpump.yaml

```diff
@@ -76,6 +76,13 @@ primary_entity:
       name: unknown_107
       type: boolean
 secondary_entities:
+  - entity: sensor

```

Comment:
This would be better as:
```yaml
  - entity: binary_sensor
    class: problem
    category: diagnostic
    dps:
      - id: 15
        type: bitfield
        name: sensor
        mapping;
          - dps_val: 0
            value: false
          - value: true
      - id: 15
        type: bitfield
        name: fault_code
```

Reply:
Oh, yes that would be better, until I can come up with a way to map the fault codes to actual meaningful messages. I know that 4 equals Error E3 which is insufficient water flow. However, all other possible errors are unknown. I'll try to contact a manufacturer of the heatpump to see if we can get some more info.

Reply:
For now, I've updated the file with your changes

Reply:
OK, I found some additional info. Can you tak a look at https://github.com/make-all/tuya-local/discussions/2154 ? The heatpump sends out bit flags for the errors.